dms/models/directory.py

```diff
@@ -691,7 +650,9 @@ def create(self, vals_list):
                 parent = self.browse([vals["parent_id"]])
                 data = next(iter(parent.sudo().read(["storage_id"])), {})
                 vals["storage_id"] = self._convert_to_write(data).get("storage_id")
-        return super().create(vals_list)
+        res = super(DmsDirectory, self.sudo()).create(vals_list)
+        res.invalidate_cache()
+        return res
```

Comment:
This has some bugs, and also makes me wonder some things.

1. You're returning a sudoed recordset. That's *very* dangerous.
2. When you resudo it back to the current user, probably the cache will be invalidated automatically (because the cache is based on its env, and sudo creates a new env). However, don't trust me and make sure that's how it works.
3. If I'm wrong in the previous point and you really need to invalidate cache, then you also need to put a comment explaining why that's necessary. Also, you should pass a list of fields to invalidate, because only the permission fields are relevant for this use case.
4. You can't just use sudo and go on without putting a comment explaining why you're doing what you're doing. Be generous on comments, please (actually lack of comments made this refactor very hard 😅).

So, probably the fix would be close to this, but please double-check it:

```suggestion
        # Create as sudo to avoid testing creation permissions before DMS security
        # groups are attached (otherwise nobody would be able to create)
        res = super(DmsDirectory, self.sudo()).create(vals_list)
        # Go back to original user and check we really had creation permission
        res = res.sudo(self.env.user)
        res.check_access_rights("create")
        res.check_access_rule("create")
        return res
```